### PR TITLE
(fix): Add some missing std::moves to C10

### PR DESCRIPTION
**Pull request by Skylion007**
I saw some missed optimization opportunities in C10 using std::move and thought I would submit a PR to fix them. There are particularly a lot of them dealing with the symbolic operators which are used in quite a few places including in loops.

### CODE REVIEW

**Comment by Skylion007** [comment]
@ezyang You seem to have been messing around with the C10 Scalar code, maybe you would be a good reviewer for this PR?

**Comment by Skylion007** [comment]
@ezyang Sounds good.

File: c10/core/TensorImpl.cpp [hunk 1]
```diff
@@ -770,7 +772,7 @@ void TensorImpl::Extend(int64_t num, float growthPct) {
   SizesVector newDims(sizes_and_strides.begin(), sizes_and_strides.end());
   newDims[0] += num;
   if (!storage_.data()) {
-    Resize(newDims);
+    Resize(std::move(newDims));
```
Comment (ezyang) [inline c10/core/TensorImpl.cpp:775]: Cannot do this, newDims is used afterwards. For safety let's not touch the legacy Caffe2 operations

**Comment by ezyang** [comment]
Looks good, just need to fix the caffe2 problem

File: c10/core/TensorImpl.cpp (hunk 1 above)
Comment (Skylion007) [reply c10/core/TensorImpl.cpp:775]: Huh? How is it used afterwards? newDims is a local variable and the next line is a void return statement and therefore the dtor on newDims is immediately called. I'll make this change anyway though. newDims is used later, but not on this if branch.

Comment (ezyang) [reply c10/core/TensorImpl.cpp]: oh oops, I missed the return, my bad!

**Review by ezyang — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**